src/wallet/adapters/ArweaveAdapter.ts

```diff
@@ -20,13 +20,15 @@ export class ArweaveAdapter implements WalletAdapter {
       this.address = await arweaveInstance.wallets.jwkToAddress(walletJWK)
     } else {
       this.signer = 'use_wallet'
+      await window.arweaveWallet.connect(['ACCESS_ADDRESS', 'ACCESS_PUBLIC_KEY', 'SIGNATURE', 'SIGN_TRANSACTION'])
```

Comment:
Added connect with required permissions: `ACCESS_ADDRESS`, `ACCESS_PUBLIC_KEY`, `SIGNATURE` and `SIGN_TRANSACTION`.